pkg/scheduler/framework/v1alpha1/registry.go

```diff
@@ -20,28 +20,18 @@ import (
 	"fmt"
 
 	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/apimachinery/pkg/util/json"
 	"sigs.k8s.io/yaml"
 )
 
 // PluginFactory is a function that builds a plugin.
-type PluginFactory = func(configuration *runtime.Unknown, f FrameworkHandle) (Plugin, error)
+type PluginFactory = func(configuration *runtime.RawExtension, f FrameworkHandle) (Plugin, error)
 
 // DecodeInto decodes configuration whose type is *runtime.Unknown to the interface into.
-func DecodeInto(configuration *runtime.Unknown, into interface{}) error {
+func DecodeInto(configuration *runtime.RawExtension, into interface{}) error {
 	if configuration == nil || configuration.Raw == nil {
 		return nil
 	}
-
-	switch configuration.ContentType {
-	// If ContentType is empty, it means ContentTypeJSON by default.
-	case runtime.ContentTypeJSON, "":
-		return json.Unmarshal(configuration.Raw, into)
-	case runtime.ContentTypeYAML:
-		return yaml.Unmarshal(configuration.Raw, into)
-	default:
-		return fmt.Errorf("not supported content type %s", configuration.ContentType)
-	}
+	return yaml.Unmarshal(configuration.Raw, into)
```

Comment:
are we saying the configuration should always be yaml? is this something k8s assumes about component config in general?

Reply:
or is this going to be serialized back by the scheduler's initialization logic (so we can always make sure it is yaml) before passing it to plugin initialization (as discussed in option 2 in https://github.com/kubernetes/kubernetes/issues/87784#issuecomment-592709524)?

Reply:
According to the RawExtension doc, scheme's Decode will already parse into JSON from json or yaml.
I just added a test for this in `cmd/kube-scheduler/app/options/options_test.go`

Reply:
I am not sure I understand, this line assumes the raw data is always yaml, why is that ok?

Reply:
Oh, forgot to mention that JSON is valid YAML. Maybe we should be parsing with json instead. But I don't think it's a big deal. For in-tree plugins we will actually depart from using this function altogether and rely on Decode only. I'm working on the next PR that does that.

That said, I don't know if Decode is able to parse any other formats.

@liggitt, anything to comment here?

Reply:
Can you clarify the approach you are taking? Is it option 2 in https://github.com/kubernetes/kubernetes/issues/87784#issuecomment-592709524? I don't remember we finalized the discussion, right?

Reply:
I added the plan here https://github.com/kubernetes/kubernetes/issues/87784#issuecomment-610393536

Reply:
There is a utility which can detect the format automatically and then use the corresponding decoder to decode it:

https://github.com/kubernetes/kubernetes/blob/3c3fc800df015955d7c38f7b39c94f3d261fd8d7/staging/src/k8s.io/apimachinery/pkg/util/yaml/decoder.go#L207

Reply:
We will be using the UniversalDecoder soon.